### PR TITLE
Nate/feature/LOOP-1419/loop status report

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-1419

Displays bluetooth status highlight based on the known state.

### CODE REVIEW

**Review by rickpasetto — COMMENTED** [review]
One overall comment: it seems to me that this is an odd mixture of responsibilities -- to have `LoopAlertsManager` also report on Bluetooth State.  I think what would feel better to me would be to create a separate class, `BluetoothManager` (or maybe `BluetoothStateManager`) which had a `BluetoothStateObserver`, which then was _used by_ `LoopAlertsManager`.  I would expect this would be just moving around of this code (and some renaming), but other than that the functionality would remain the same.   WDYT?

(I know we had chatted about this before, and I'm sorry if this is somewhat of a retraction of what I told you...)

**Comment by nhamming** [comment]
> One overall comment: it seems to me that this is an odd mixture of responsibilities -- to have `LoopAlertsManager` also report on Bluetooth State. I think what would feel better to me would be to create a separate class, `BluetoothManager` (or maybe `BluetoothStateManager`) which had a `BluetoothStateObserver`, which then was _used by_ `LoopAlertsManager`. I would expect this would be just moving around of this code (and some renaming), but other than that the functionality would remain the same. WDYT?
> 
> (I know we had chatted about this before, and I'm sorry if this is somewhat of a retraction of what I told you...)

Yeah, I'm all for this and will make the update. Only item that comes to mind now, is that the app delegate seems like it will be getting crowded.

**Review by ps2 — APPROVED** [review]
I agree with @rickpasetto that keeping bluetooth status watching code in it's own class makes sense. Another responsibility simplification could be to reduce the StatusViewController's responsibility for listening to and sorting out the different types of StatusHighlights, and just make the AlertManager vend an "alertStatusHighlight", that the status viewcontroller would subscribe to.

**Comment by rickpasetto** [comment]
I like Pete's suggestion. +1

**Comment by nhamming** [comment]
A refactor that includes the suggestion, but not fully. 

I think it makes sense for the StatusTableViewController to know the bluetooth state direct the user to correct the issue.

@ps2 I know this doesn't fully address your comment. Is it a suitable alternative? Or should I continue to reduce the complexity of that VC?

**Review by ps2 — APPROVED** [review]
I still think it'd be nice for StatusViewController to just be responsible for showing the highlight, and responding to user taps on the highlight in a generic sense by letting the issuer of the highlight know about the tap. The issuer can then be responsible for dealing with whatever the next action is. But I won't block this; the code looks to do what it intends to do, and is ok as is.

File: Loop/AppDelegate.swift [hunk 1]
```diff
@@ -35,7 +36,10 @@ final class AppDelegate: UIResponder, UIApplicationDelegate {
         alertManager = AlertManager(rootViewController: rootViewController, expireAfter: Bundle.main.localCacheDuration ?? .days(1))
         deviceDataManager = DeviceDataManager(pluginManager: pluginManager, alertManager: alertManager)
         loopAlertsManager = LoopAlertsManager(alertManager: alertManager)
-
+        bluetoothStateManager = BluetoothStateManager()
```
Comment (rickpasetto) [inline Loop/AppDelegate.swift:38]: [nit] I would have expected that `BluetoothStateManager` would be injected into `LoopAlertsManager` and then `LoopAlertsManager` would add _itself_ as an observer, but that's a minor point.

File: Loop/View Controllers/StatusTableViewController.swift [hunk 2]
```diff
@@ -499,22 +508,30 @@ final class StatusTableViewController: ChartsTableViewController {
...
                                                             sensor: self.deviceManager.sensorState)
-
-                    if self.deviceManager.cgmManager != nil {
-                        hudView.cgmStatusHUD.presentStatusHighlight((self.deviceManager.cgmManager as? CGMManagerUI)?.cgmStatusHighlight)
-                    }
                 }
                 
-                // Pump Status HUD
-                if self.deviceManager.pumpManager != nil {
+                if let bluetoothStatusHighlight = self.bluetoothState.statusHighlight {
```
Comment (rickpasetto) [inline Loop/View Controllers/StatusTableViewController.swift:521]: [nit] seems management of the data shown by the HUDView would be in a _ViewModel_, not here in `StatusTableViewController`...but that's probably for a later refactor/cleanup

**Review by rickpasetto — APPROVED** [review]
Some nits but not blocking at all.  LGTM

File: Loop/View Controllers/StatusTableViewController.swift (hunk 2 above)
Comment (nhamming) [reply Loop/View Controllers/StatusTableViewController.swift:521]: While I agree, I'll leave this for a future refactor.

**Pull request merged**